### PR TITLE
add support for the aws cloud plugin

### PR DESCRIPTION
**Pull request by harishneit**
this plugin enables automatic disovery of elasticsearch nodes based on security group and the tag values provided.

when configured with a bucket, this plugin enables saving snapshots of the indices and restore indices from snapshots on s3.

Documentation for the plugin is available at: https://www.elastic.co/guide/en/elasticsearch/plugins/2.3/cloud-aws.html

### CODE REVIEW

File: templates/etc/elasticsearch/elasticsearch.yml.j2 [hunk 1]
```diff
@@ -441,3 +441,63 @@ http.jsonp.enable: {{ elasticsearch_jsonp_enabled | lower }}
...
+
+{% if elasticsearch_s3_bucket_name %}
+repositories.s3.bucket: {{elasticsearch_s3_bucket_name}}
+{% endif %}
+
+{% if elasticsearch_plugin_aws_region %}
+repositories.s3.region: {{elasticsearch_plugin_aws_region}}
+{% endif %}
+
+{% if elasticsearch_s3_bucket_name %}
```
Comment (rishabh-g3) [inline templates/etc/elasticsearch/elasticsearch.yml.j2]: the variable is already declared above..

File: templates/etc/elasticsearch/elasticsearch.yml.j2 [hunk 2]
```diff
@@ -441,3 +441,63 @@ http.jsonp.enable: {{ elasticsearch_jsonp_enabled | lower }}
...
+
+
+########################### EC2 Discovery Configs ###########################
+
+{% if elastsearch_plugin_mandatory %}
+plugin.mandatory: {{elastsearch_plugin_mandatory}}
+{% endif %}
+
+{% if elastsearch_node_ip %}
+network.publish_host: {{elastsearch_node_ip}}
```
Comment (rishabh-g3) [inline templates/etc/elasticsearch/elasticsearch.yml.j2]: pls put this variable assignment in if-else block b/c this variable is already declared above which has a default value of "0.0.0.0" when used with ansible repo...

File: templates/etc/elasticsearch/elasticsearch.yml.j2 (hunk 1 above)
Comment (harishneit) [inline templates/etc/elasticsearch/elasticsearch.yml.j2]: @rishabh-g3 I have fixed it in [pull#7](https://github.com/grofers/ansible-elasticsearch/pull/7).

File: templates/etc/elasticsearch/elasticsearch.yml.j2 (hunk 2 above)
Comment (harishneit) [inline templates/etc/elasticsearch/elasticsearch.yml.j2]: shouldn't we not publish 0.0.0.0 as host ip?

**Pull request closed without merging**